### ISSUE TITLE
Pagination position is retained between searches

### ISSUE THREAD

**Issue by ybnd** [DSpace/dspace-angular#2159]
**Describe the bug**
The current page of the search results component is retained between searches.
This is not a breaking bug when going between queries that have roughly the same amount of results, but is confusing nevertheless.
When going from a query with many results to one with only a few you get a "query has no results" message because the position does not reset to the first page.
In this case the UI does not communicate that you're trying to show e.g. page 10 of 2.

**To Reproduce**
1. Open https://demo7.dspace.org/search
2. Go to the last results page
3. Change the query to `test` → bound to have fewer results than _everything_
   * "Your search returned no results"
   * The pagination UI is not shown, the only way to find out you're on a page that doesn't exist is through the URL: it will still have e.g. `spc.page=170`

**Expected behavior**
Each time a new search is performed, the UI should reset to the first page.

**Related work**
Link to any related tickets or PRs here.

**Comment by tdonohue** [comment]
Agreed this is an annoying bug. Pulling over to 7.6 board to see if we can find a volunteer

**Comment by CVamsi27** [comment]
I was not able to reproduce the issue.
- I started off with [https://demo7.dspace.org/search](https://demo7.dspace.org/search)
- Navigate to page 596 
![image](https://user-images.githubusercontent.com/55316723/232278252-1a04c629-53b0-4a36-877e-f6ab0ac0c8e5.png)
- Now change the URL to [https://demo7.dspace.org/search?query=Test](https://demo7.dspace.org/search?query=Test)
- Now I can see its back to page 1
![image](https://user-images.githubusercontent.com/55316723/232278355-017b8535-3cf2-4db5-af6c-1cc75065f6e1.png)

**Comment by tdonohue** [comment]
@CVamsi27 & @ybnd : I am able to reproduce this bug on demo7.dspace.org and locally... but the steps I used are slightly different:

1. Go to https://demo7.dspace.org/search  (empty search)
2. Click on last page of results. As of today, that's page 609 (it may be different when others test if content is added/removed).  After clicking on that last page of results, the URL in your browser window will change to be: `https://demo7.dspace.org/search?spc.page=[last-page-number]`
3. Now, go to the searchbox in the middle of the page & enter "test" and click Search.  The results will display "Your search returned no results".  The reason is that `spc.page` param still exists in the URL of your browser window... you'll now see `https://demo7.dspace.org/search?spc.page=[last-page-number]&query=test`   If you remove that `spc.page` param from the browser URL, the page will reload and show you the results for the "test" query.

I believe you didn't do the same step 3 as I did.  It sounds like you changed the browser URL manually (which works fine)... the bug is only reproducible if you enter a new search term in the search box and click "search"
